### ISSUE TITLE
transmission uses GB instead of GiB

### ISSUE THREAD

**Issue by ewtoombs** [transmission/transmission#198]
Yes, it's a BUG! NOBODY expects GB! EVERYBODY expects GiB!

**Comment by timmerk** [comment]
Not a bug. I just asked a bunch of people, and no one knows what a GiB is.

**Comment by mikedld** [comment]
Related to [TRAC-4703](https://trac.transmissionbt.com/ticket/4703), [TRAC-4747](https://trac.transmissionbt.com/ticket/4747).

**Comment by ewtoombs** [comment]
They may not know what the letters GiB mean, but they do know what a GiB is. GiB is the unit they are familiar with. A GB is considerably smaller than a GiB and is almost never used. Except by Transmission. Transmission is telling people the files it is downloading are bigger than they actually are.

**Comment by timmerk** [comment]
All Mac apps use "GB", not "GiB". Chrome uses GB. Firefox uses GB. Mac OS X uses GB. Windows uses GB. And so on.

**Comment by ewtoombs** [comment]
I'm not talking about the spelling. Spell it however you want to. I'm talking about the unit itself. Transmission can keep labelling it GB if they want to.

**Comment by ewtoombs** [comment]
And every one of those programs uses the GiB. They spell it GB, but they're using the GiB—not the GB. Almost nobody uses it.

**Comment by timmerk** [comment]
Ohh, ok, ok. I agree with you then. Keep spelling as "GB" but use GiB as actual unit.

**Comment by ewtoombs** [comment]
Look you really need to actually look these things up before saying anything. http://bfy.tw/AQYJ

**Comment by timmerk** [comment]
No, you weren't clear that you were talking about just the units, but not the actual text.

**Comment by ewtoombs** [comment]
Oh right I see what you mean. Yeah, Transmission really is doing it wrong. They aren't just labelling it wrong.

**Comment by ewtoombs** [comment]
Other clients do use the right label. See for example, qbittorrent.

**Comment by xelra** [comment]
That would actually be really, really bad. If you want GiB, you write GiB. If you want GB, you write GB. **But don't mislabel GiB as GB!**

**Comment by ewtoombs** [comment]
I agree, xelra, it is better to write GiB. It is common practice, though, to write GB, so if the makers are concerned about confusing their users, I'd understand if they decided to keep it that way. They definitely need to start using the GiB unit in their size calculations, though.

**Comment by jorvi** [comment]
Transmission should use GiB on Linux and GB on macOS. macOS switched from GiB to GB back with 10.6 Snow Leopard. Since the OS itself and most (updated) apps use GB, Transmission should stick to convention.

**Comment by killemov** [comment]
We have hard-drive manufacturers to thank for this distinction. A kilobyte used to be 1024 bytes, no questions asked. They opted to go metric and make it 1000 bytes. Now kB is 1000 and kiB is 1024.

**Comment by RuthlessRuler** [comment]
Any Updates on Implementing this?

**Comment by gingerbeardman** [comment]
I would prefer Transmission to display Kilobits/Megabits rather than Kilobytes/Megabytes, so I can more easily work with my allocated 50Megabit broadband.

4500KB/s = 36Megabits/s

![Screen shot 2019-12-19 at 11 32 29](https://user-images.githubusercontent.com/49612/71176217-44394f00-2261-11ea-82a2-3d79dbc8dedb.png)

**Comment by Coeur** [comment]
Regarding macOS, the code uses the "**File**" style:
https://github.com/transmission/transmission/blob/d3e1c83c9b348c2ee2538498f9938ded16fea1a5/macosx/NSStringAdditions.mm#L40
From the doc, there are four possible styles:
https://developer.apple.com/documentation/foundation/bytecountformatter/countstyle
>.file uses the .decimal style, but that may change over time.
>.memory uses the .binary style, but that may change over time.
>.decimal Causes 1000 bytes to be shown as 1 KB.
>.binary Causes 1024 bytes to be shown as 1 KB.

So... in all scenarios, Apple will display "GB", not "GiB".
We can either leave it like that, wait for Apple to change its "File" style preferences everywhere, or adopt the Binary style to change the maths ourselves (but still displaying as GB). Or do some more complex localisation overrides ourselves.

**Comment by GaryElshaw** [comment]
I asked about this a while back, i'd love the option to have MB instead of GB to help get better insight into how much time might be necessary for smaller files.
